src/Workspaces/Core/Portable/Execution/AssetStorages.Storage.cs

```diff
@@ -33,6 +35,11 @@ public Storage(SolutionState solutionState)
                 _serializer = SolutionState.Workspace.Services.GetRequiredService<ISerializerService>();
             }
 
+            public void Dispose()
+            {
+                _keepAliveObjects.Clear();
+            }
```

Comment:
i don't get this Dispose.  Won't Storage just be GC'ed, which will release _kepeAliveObjects, which will release anythin that points at?

So all that Clear does is make those pointed-at objects elegible for GC as wlel...  But if that's what we care about, then that would happen when Storage itself was GC'ed.

Reply:
`Storage` could be holding a fairly large working set. Clearing the rooted objects here can help with post-mortem investigations.

Reply:
This is not strictly necessary. It was originally included because `_keepAliveObjects` was an `ImmutableArray<GCHandle>`, but that was since changed to `ImmutableArray<object>`.

Reply:
> Clearing the rooted objects here can help with post-mortem investigations, and ensure that the keep alive objects can be reclaimed even if Storage remains reachable for some reason.

I can see that i suppose.  But it feels very speculative.  Seems like it would be super odd for someone to have a ref to the storage, and have RAII somewhere to dispose it, but would still be rooting it.

My stronger preference in that case would be that the current client that is explicitly disposing their ref simply null-out their ref, allowing the GC to take over here.  

> This is not strictly necessary. It was originally included because _keepAliveObjects was an ImmutableArray<GCHandle>, but that was since changed to ImmutableArray<object>.

Gotcha.  Given that change, i would still land on def preferring no actual dispose here.  And, if you do worry about clearing the rooted objects, that's done by having the clients simply releasing their Storage instances instead of .Dispose'ing them.

Reply:
➡️ Removed `IDisposable` from this type